pkg/registry/generic/etcd/etcd.go

```diff
@@ -459,16 +462,56 @@ func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, lis
 	if err != nil {
 		return nil, err
 	}
-	for _, item := range items {
-		accessor, err := meta.Accessor(item)
-		if err != nil {
-			return nil, err
-		}
-		if _, err := e.Delete(ctx, accessor.GetName(), options); err != nil {
-			return nil, err
+	// Spawn a number of goroutines, so that we can issue requests to etcd
+	// in parallel to speed up deletion.
+	workersNumber := registryoptions.GetDeleteCollectionParallelism()
```

Comment:
@smarterclayton - what I wanted to avoid is that basically we are passing the same number to all per-resource-etcd`s.
So basically, we would need to change all NewREST methods for all resources e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/master/master.go#L297
to take an additional argument
I don't think it's a good idea.

Another option would be to still leave it global, but inject it in per-resource etcd code similarly to what we are doing with cachesize:
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/pod/etcd/etcd.go#L71

And the last option is to just pass it explicitly in all NewREST functions, which I wanted to avoid?

@smarterclayton - do you really want to go with the last option? Or are you fine with the second option?